### PR TITLE
Add referenced BackendRefs for ExtAuth to Resource Tree

### PR DESCRIPTION
**Pull request by zhaohuabing**
ExtAuth functionality was broken due to missing BackendRefs in the Resource Tree. This PR resolves this issue.
I have manually verified the fix in my dev environment and currently working on the associated e2e test. The e2e test will be in a separate follow-up PR.

This PR also adds BackendTLSPolicy support to the BackendRefs of the Ext Auth services.

### CODE REVIEW

File: internal/xds/translator/extauth.go
Comment (arkodg) [inline internal/xds/translator/extauth.go:97]: why not use `AUTO` here ?

File: internal/provider/kubernetes/controller.go
Comment (arkodg) [inline internal/provider/kubernetes/controller.go:430]: can you make sure that if a backendTLSPolicy exists for this backendRef that is included into `resources` as well ?

sidenote (not directly related to this PR) I dont see any tls context in the translation https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/gatewayapi/securitypolicy.go#L753
shouldnt we be reusing this function ?
https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/gatewayapi/route.go#L1074

File: internal/xds/translator/extauth.go
Comment (zhaohuabing) [reply internal/xds/translator/extauth.go:97]: It's deprecated. We should set it to v3 explicitly.
```
	// When not specified, we assume v2, to ease migration to Envoy's stable API
	// versioning. If a client does not support v2 (e.g. due to deprecation), this
	// is an invalid value.
	//
	// Deprecated: Marked as deprecated in envoy/config/core/v3/config_source.proto.
	ApiVersion_AUTO ApiVersion = 0
```

File: internal/provider/kubernetes/controller.go
Comment (zhaohuabing) [reply internal/provider/kubernetes/controller.go:430]: > shouldnt we be reusing this function ?

I have a TODO in the code and plan to do that but maybe later after v1.0.0. It requires massive modification on the current xRoute translation because the current Destination processing is tied to RouteParentContext.

File: internal/provider/kubernetes/controller.go
Comment (arkodg) [inline internal/provider/kubernetes/controller.go:373]: you probably need to also make sure that the controller reconciles if the ext auth backend ref changes :)
so the indexer needs to be added or updated similar to https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/provider/kubernetes/indexers.go#L380

Comment (zhaohuabing) [reply internal/provider/kubernetes/controller.go:373]: @arkodg  It's already there https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/provider/kubernetes/indexers.go#L406

File: internal/gatewayapi/backendtlspolicy.go
Comment (zhaohuabing) [inline internal/gatewayapi/backendtlspolicy.go:166]: Why do we need to check whether the parent Gateway exist?

**Review by arkodg — APPROVED** [review]
LGTM thanks !
its a big change :). but the ext auth e2e and backend tls e2e should help verify this

**Comment by arkodg** [comment]
e2e is failing, kicked the CI, lets wait for the tests to pass before we merge

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**
